### PR TITLE
Add database_index for redis instrumentation

### LINKED ISSUE #209: Add database_index attribute for redis instrumentation when db is not 0
Prior discussion [here](https://github.com/open-telemetry/opentelemetry-python-contrib/pull/184#discussion_r528913282).

The spec says for redis if db is not 0 it should have `db.redis.database_index` set to the index of the database being accessed as used in the SELECT command, provided as an integer.

### PR DESCRIPTION
**Pull request by lonewolf3739**
Fixes #209

### CODE REVIEW

**Comment by lonewolf3739** [comment]
cc @lzchen @NathanielRN

File: tests/opentelemetry-docker-tests/tests/redis/test_redis_functional.py [hunk 1]
```diff
@@ -115,3 +115,31 @@ def test_parent(self):
...
+    def _check_span(self, span, name):
+        self.assertEqual(span.name, name)
+        self.assertIs(span.status.status_code, trace.status.StatusCode.UNSET)
+        self.assertEqual(span.attributes["net.peer.name"], "localhost")
+        self.assertEqual(span.attributes["net.peer.ip"], 6379)
+        self.assertEqual(span.attributes["db.redis.database_index"], 10)
+        self.assertNotIn("db.name", span.attributes)
+
+    def test_get(self):
+        self.assertIsNone(self.redis_client.get("paneer"))
```
Comment (NathanielRN) [inline tests/opentelemetry-docker-tests/tests/redis/test_redis_functional.py:140]: Lol, I don't mind `paneer` at all but `foo` might convey that it's just a random string better 😝

**Review by NathanielRN — APPROVED** [review]
Sweet thank you for the change! Just some non-blocking comments!

File: instrumentation/opentelemetry-instrumentation-redis/src/opentelemetry/instrumentation/redis/util.py [hunk 2]
```diff
@@ -21,8 +21,12 @@ def _extract_conn_attributes(conn_kwargs):
     """ Transform redis conn info into dict """
     attributes = {
         "db.system": "redis",
-        "db.name": conn_kwargs.get("db", 0),
     }
+    db = conn_kwargs.get("db", 0)
+    if db == 0:
+        attributes["db.name"] = db
+    else:
+        attributes["db.redis.database_index"] = db
```
[the remark this answers is not in the document]
Comment (lonewolf3739) [reply instrumentation/opentelemetry-instrumentation-redis/src/opentelemetry/instrumentation/redis/util.py:29]: Spec says it is conditionally required. May be we can set both `db.name` and `db.redis.database_index`? Let others also share opinions.

File: tests/opentelemetry-docker-tests/tests/redis/test_redis_functional.py (hunk 1 above)
Comment (lonewolf3739) [reply tests/opentelemetry-docker-tests/tests/redis/test_redis_functional.py:140]: Prev tests had `cheese` so just changed it to another related food name here. No worries. I will change this to `foo`.

File: instrumentation/opentelemetry-instrumentation-redis/src/opentelemetry/instrumentation/redis/util.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (lzchen) [reply instrumentation/opentelemetry-instrumentation-redis/src/opentelemetry/instrumentation/redis/util.py:29]: My vote is for both.

**Review by lzchen — APPROVED** [review]

**Pull request merged**